### PR TITLE
Added training and inference metric + seed arg

### PR DESCRIPTION
**Pull request by msaroufim**
Logs look like this now - this will be useful to work through numerical stability issues

```
Training finished.
First iteration took: 25.94s
Average time after the first iteration: 117.50ms
Training Accuracy for backend inductor: {'accuracy': 0.7579148471615721, 'f1': 0.831015431510764}
Evaluation finished.
First iteration took: 9.08s
Average time after the first iteration: 36.98ms
Test Accuracy for backend inductor: {'accuracy': 0.8475, 'f1': 0.8935427574171029}
```

### CODE REVIEW

File: scripts/text_classification.py [hunk 1]
```diff
@@ -223,7 +234,7 @@ def preprocess_function(examples):
             outputs = model(**batch)
         predictions = outputs.logits.argmax(dim=-1) if not is_regression else outputs.logits.squeeze()
         predictions, references = accelerator.gather_for_metrics((predictions, batch["labels"]))
```
Comment (msaroufim) [inline scripts/text_classification.py]: not 100% sure if this is correct way to use gather

Comment (pacman100) [reply scripts/text_classification.py]: Hello, `gather_for_metrics` takes `torch.Tensor, or a nested tuple/list/dictionary of torch.Tensor` as input. So, this is correct usage.

**Review by sgugger — COMMENTED** [review]
You don't need two `metric` objects, you can reuse the first one as doing `metric.compute()` will reset its state. Apart from that LGTM!

**Review by sgugger — APPROVED** [review]
Thanks!

Side note: the training metric is computed over all epochs, not just one. Shouldn't be a problem for metrics that are averages, but you might want to put the `metric.compute()` inside the for loop of epochs.
Merging for now in any case!
